app/components/record_actions/view.rb

```diff
@@ -45,5 +55,17 @@ def withdraw_link
     def reinstate_link
       govuk_link_to(t("views.trainees.edit.reinstate"), trainee_reinstatement_path(trainee), class: "reinstate")
     end
+
+    def trainee_course_start_date_in_future?
+      trainee.course_start_date && trainee.course_start_date > Time.zone.now
+    end
+
+    def course_started_but_trainee_has_not_specified_start_date?
+      course_start_date && course_start_date < Time.zone.now && trainee.course_start_date.blank?
```

Comment:
I think this should (also?) check against `trainee.commencement_date`/"trainee start date", as opposed to the course start date.

Reply:
You're right. I completely forgot about `commencement_date`. Nice catch.